ros2_ws/src/rcdt_utilities/rcdt_utilities/test_utils.py

```diff
@@ -0,0 +1,135 @@
+from rclpy.action import ActionClient
+from sensor_msgs.msg import JointState
+from std_srvs.srv import Trigger
+from trajectory_msgs.msg import JointTrajectoryPoint
+
+
+def get_joint_position(name: str) -> float:
+    """Get the joint position of a joint by name. This is done by calling the
+    /joint_states topic and parsing the output. This is a workaround for the fact
+    that the joint states are not published in a format that can be easily parsed.
+    Also --field does not work with the /joint_states topic.
+
+    Args:
+        name (str): The name of the joint.
+    Returns:
+        float: The position of the joint.
+    """
+    topic_list = [("joint_states", JointState)]
+    wait_for_topics = WaitForTopics(topic_list, timeout=10.0)
+    assert wait_for_topics.wait()
+    msg = wait_for_topics.received_messages("joint_states")[0]
+    idx = msg.name.index(name)
+    position = msg.position[idx]
+    wait_for_topics.shutdown()
+    return position
+
+
+def call_trigger_service(node: rclpy.node.Node, service_name: str) -> bool:
+    """Call a trigger service and return True if the service was called successfully."""
+    client = node.create_client(Trigger, service_name)
+    if not client.wait_for_service():
+        raise RuntimeError(f"Service {service_name} not available")
+
+    future = client.call_async(Trigger.Request())
+    rclpy.spin_until_future_complete(node, future=future)
+    return future.result() is not None
+
+
+def call_move_to_configuration_service(
+    node: rclpy.node.Node, configuration: str = "drop"
+) -> bool:
+    """Call the move_to_configuration service and return True if the service was called"""
+    client = node.create_client(
+        MoveToConfiguration, "/moveit_manager/move_to_configuration"
+    )
+    if not client.wait_for_service():
+        raise RuntimeError(
+            "Service /moveit_manager/move_to_configuration not available"
+        )
+
+    request = MoveToConfiguration.Request()
+    request.configuration = configuration
+    future = client.call_async(request)
+    rclpy.spin_until_future_complete(node, future=future)
+    return future.result() is not None
+
+
+def call_move_gripper_service(
+    node: rclpy.node.Node, width: float, action_name: str
+) -> bool:
+    """Call the gripper to go to a defined width."""
+    action_client = ActionClient(node, GripperCommand, action_name=action_name)
+
+    if not action_client.wait_for_server():
+        raise RuntimeError(f"Action server {action_name} not available")
+
+    goal_msg = GripperCommand.Goal()
+    goal_msg.command.position = width
+
+    future = action_client.send_goal_async(goal_msg)
+
+    rclpy.spin_until_future_complete(node, future)
+    goal_handle = future.result()
+    if not goal_handle.accepted:
+        return False
+
+    result_future = goal_handle.get_result_async()
+    rclpy.spin_until_future_complete(node, result_future)
+    result = result_future.result().result
+    return result.reached_goal
```

Comment:
I usually try to define all types if not done automatically by the used methods:


```suggestion
    goal_handle: ClientGoalHandle = future.result()
    print(type(goal_handle))
    if not goal_handle.accepted:
        return False

    result_future: rclpy.task.Future = goal_handle.get_result_async()
    rclpy.spin_until_future_complete(node, result_future)
    result: GripperCommand.Impl.GetResultService.Response = result_future.result()
    return result.result.reached_goal
```

We can discuss if we want to follow this!